src/api/pubsub.js

```diff
@@ -0,0 +1,105 @@
+'use strict'
+
+const promisify = require('promisify-es6')
+const bs58 = require('bs58')
+var Base64 = require('js-base64').Base64
+var Stream = require('stream')
+var http = require('http')
+
+let activeSubscriptions = []
+
+const subscriptionExists = (subscriptions, topic) => {
+  return subscriptions.indexOf(topic) !== -1
+}
+const removeSubscription = (subscriptions, topic) => {
+  const indexToRemove = subscriptions.indexOf(topic)
+  return subscriptions.filter((el, index) => {
+    return index !== indexToRemove
+  })
+}
+const addSubscription = (subscriptions, topic) => {
+  return subscriptions.concat([topic])
+}
+
+module.exports = (send, config) => {
+  return {
+    sub: promisify((topic, options, callback) => {
+      if (typeof options === 'function') {
+        callback = options
+        options = {}
+      }
+      if (!options) {
+        options = {}
+      }
+
+      var rs = new Stream()
+      rs.readable = true
```

Comment:
you can use through2 instead, that way you can pipe the respons into your transform